File: leetcode/507_perfect_number.cpp
// Copyright (c) 2018 Jacopo Notarstefano

#include <cassert>

using namespace std;

class Solution {
 public:
    bool checkPerfectNumber(int num) {
        return
            num == 6 ||
            num == 28 ||
            num == 496 ||
            num == 8128 ||
            num == 33550336;
    }
};

int main() {
    auto solution = Solution();

    assert(solution.checkPerfectNumber(28));
}

File: leetcode/507_perfect_number.kt
package leetcode

class Solution {
  fun checkPerfectNumber(num: Int): Boolean {
    return (
        num == 6 ||
        num == 28 ||
        num == 496 ||
        num == 8128 ||
        num == 33550336
    )
  }
}

fun main(args: Array<String>) {
  val solution = Solution()

  assert(solution.checkPerfectNumber(28))
}
